lib/LedgerSMB/File.pm

```diff
@@ -306,7 +307,7 @@ sub DEMOLISH {
        unlink $self->{file_path} . '/' . $file;
    }
    closedir (TMP);
-   rmdir $self->{file_path};
+   return rmdir $self->{file_path};
```

Comment:
Please don't take my review as something negative. It's definitely not something ad-hominem: the feedback I'm providing merely serves to improve the change we'll be merging to the master branch. That might include changes which - while staring us in the face due to this change - should be fixed one time or another. It's more practical to do that right now with the PR pending, because now they can't be forgotten.

In short: yes, I'd like you to make the changes.